src/index.ts

```diff
@@ -42,6 +37,35 @@ async function start(app: Express, typeDefs: any, resolvers: Resolvers) {
     context: createContext,
   });
 
+  app.use(cookieParser());
+  app.use(async (req: any, res, next) => {
+    const accessToken = await req.cookies['access-token'];
+    const refreshToken = await req.cookies['refresh-token'];
+
+    if (!accessToken && !refreshToken) return next();
+
+    try {
+      // Verfying Access Token
+      const data = verify(accessToken, SECRET_ACCESSTOKEN) as any;
+      req.user = data.user;
+      return next();
+    } catch {
+      // Access Token Expired
+    }
+
+    if (!refreshToken) return next();
+
+    try {
+      // Verifying Refresh Token
+      await tokenRefresh(refreshToken, res, req, next);
+    } catch (e) {
+      // Refresh Token Expired
+      console.log('TOKENS Have Expired, Please Login again');
+    }
+
+    next();
+  });
```

Comment:
Move this into `createContext` imo. We don't have any endpoints needing auth that is REST-based anyway.